### ISSUE TITLE
How use promise in new interceptors format (version 1.4.0)?

### ISSUE THREAD

**Issue by KhristenkoYura** [pagekit/vue-resource#675]

**Comment by stgogm** [comment]
Having the same issue here. Have you found a solution, @KhristenkoYura ?

**Comment by ver1000000** [comment]
Spend all day and couldn't find similar solution without `next()` method calling.

The algorithm was:
1. Check token validity before each request
2. Stop all new requests if token expired (make queue array `[{ request, next }, ...]`)
3. Wait new token
4. Release queue

But now we can't stop new requests and wait for a new token.

**Comment by stgogm** [comment]
I've finally given up and started using [vue-axios](https://github.com/imcvampire/vue-axios). It behaves the way you'd expect and is pretty much a drop-in replacement.

This project has poor documentation and with things like this make it unusable in real scenarios.

**Comment by KhristenkoYura** [comment]
@stgogm , I'm using version 1.3

**Comment by stgogm** [comment]
@KhristenkoYura Yes, I was too, but having to do an extra request when you can refresh the token beforehand if needed is unnecessary and not optimal. Axios allows you to do this by being able to add an async or Promise interceptor.

**Comment by star1105** [comment]
I'm using version 1.5.1
